### DISCUSSION TITLE
Running on Multiple Windows Accounts and Battery Limitation

### DISCUSSION THREAD

**Discussion by PetrasekMartin** [seerge/g-helper#1587] [Q&A]
When I install Ghelper on one Windows account, it runs smoothly on that account. However, if I try to login on another account, it don't run automacitly, and when i run it manuly it reports that it's already running but only starts after this message. Additionally, the battery limitation feature doesn't seem to work, as the battery always charges to 100%. It appears that the settings might be getting overwritten when I use it in this manner.

Has anyone else experienced this problem with Ghelper? Is there a solution or workaround to ensure it functions correctly on multiple Windows accounts?

Thank you for your assistance!

**Comment by seerge** [comment]
@PetrasekMartin hello, if you switch account I assume windows doesn't terminate currently running processes, so you get a notification that app is already running. When you relaunch app - it will kill other ghelper.exe process tho.

If you want to make app start with each account - you need to manually copy startup task for the app in windows Task Manager for each account

As for battery limit, you need to stop asus services, [it's already covered in FAQ](https://github.com/seerge/g-helper#battery-charge-limiter-is-not-working)

**Reply by Hal-Jordan2814** [reply]
I did this but it asks for admin privileges each time and it gets annoying after a while. is there a way to do it so it works on my standart user account but doesn't ask for the password every time.

**Reply by seerge** [reply]
@Hal-Jordan2814 at what exact moment does it ask for admin permissions ?

**Reply by haljordan-2814** [reply]
İt was asking at startup every time. I managed to fix it just by going into the executable itself and gave it full control for the standard account from there. Now everything is working perfectly fine and it's not prompting me for the password.

**Reply by Hal-Jordan2814** [reply]
never mind it started to ask the password on boot again. do you know how to solve this on a standard user. It asks it on log on.

**Reply by vertner82** [reply]
I have the same issue. I have my rog flow z13 setup for my work computer. I have a personal account on it and my work account. I also have admin rights account. If I use my personal or admin account it works GHelper works good. I have to enter my admin creds when i am logged into my work account. Is there a way around this? I was hoping to use GHelper to get around the armory crate admin stuff. Is there a way to make this work with admin on all accounts?

**Reply by seerge** [reply]
@vertner82 just run app as admin at least once and it will create a separate charge setting task that runs before user login even

But if you want to use app UI from different accounts you need to manually create separate startup tasks for GHelper.exe under each account

**Reply by TomSalvador90** [reply]
Is there no way to make sure running it as admin creates the startup task? Secondary standard (non-admin) user accounts are a pretty common use case, and forcing users to fiddle with the Task Scheduler for that is probably not ideal.

**Reply by seerge** [reply]
@TomSalvador90 when you run app as admin and check `Run on Startup` - it creates a startup task with same (admin) permissions.

But problem is that due to the Windows handles UI - apps need to run under same user account as the one trying to access them. Otherwise Windows simply won't show app window (but it will keep running background)

So you need to have separate startup task for each user (unfortunately).

**Reply by TomSalvador90** [reply]
Okay, so the solution would be creating an user-specific startup task for each account? 
From my limited testing that still throws a prompt that the app is already running when **switching** users even when marking the option to not run when an instance of the task when is already present.
This makes it not seamless and it would be nice to make sure another user can have access to the app, the shortcuts and the UI when they log into the computer in a clean way, even if it's  not the admin.

**Reply by seerge** [reply]
@TomSalvador90 yes, solution would be duplicating startup task for each user (and doing this once). You can set all tasks run app with admin permissions (even if user does not have them).

So each time you log in as a new user - new process of G-Helper will start and will be accessible to the user.

**Comment by vertner82** [comment]
Serge, is there any possibility that the ghelper app can cause my airplane
mode to come on at startup and randomly turn it on every few seconds for
the first few min after startup?

**Reply by haljordan-2814** [reply]
Because Ur pc boots to quick if I press a key before GHelper is launched it constantly turns on and off aeroplane mode. Must be a bug or something. I had this before but figured it out just wait till Ur pc is fully booted up before pressing anything on the keyboard

**Reply by seerge** [reply]
@vertner82 i don’t think it’s related :) Airplane mode is a windows feature, G-Helper is not interacting with it anyhow
